PokemonRampUp/Model/NameAndURL.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  Species.swift
+//  PokemonRampUp
+//
+//  Created by Thushen Mohanarajah on 2023-02-14.
+//
+
+import Foundation
+
+// This is a generic struct that has a name and URL variable. This is generic as there is many cases where this object is found when requesting Pokemon JSON from PokeAPI
+struct NameAndURL: Decodable {
+    let name: String
+    let url: String
+    
+    init(name: String,
+         url: String
+    ){
+        self.name = name
+        self.url = url
+    }
+}
```

Comment:
What's the purpose of this `struct`? 
Can we rename this structure to give a better understanding about what "entity" it represents?

Reply:
To be honest, I am not sure what to name it. Maybe, if I explain why the struct exists you could give me a good suggestion or idea on how to come up with one. 

Explanation:

```
// This is a generic struct that has a name and URL variable. This is generic as there is many cases where this object is found when requesting a Pokemon JSON from PokeAPI

struct NameAndURL: Decodable {
    let name: String
    let url: String
    
    init(name: String,
         url: String
    ){
        self.name = name
        self.url = url
    }
}
```

`NameAndURL` is basically a generic class that is used in multiple cases. For example, take a look at this JSON blocks sent from PokeAPI:

The `moves` object in the Pokemon JSON file
```
    "moves": [
        {
            "move": {
                "name": "mega-punch",
                "url": "https://pokeapi.co/api/v2/move/5/"
            },
...
```

The `types` object in the Pokemon JSON file
```
    "types": [
        {
            "slot": 1,
            "type": {
                "name": "electric",
                "url": "https://pokeapi.co/api/v2/type/13/"
            }
        }
    ],
...

```

Do you see how there is an inner object that contains `name` and `url`. This object appears multiple times in the Pokemon JSON file. This is why I created a a model called `NameAndURL` that can be used in multiple cases.